pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -111,14 +126,55 @@ let
         cp -R --no-preserve=mode,ownership ${browser}/Applications/${browserName}.app $out/Applications
         rm -f $out${browser.execdir or "/bin"}/${browserName}
       '' + ''
-        if [ ! -x "${browser}${browser.execdir or "/bin"}/${browserName}" ]
+
+        # Link the runtime. The executable itself has to be copied,
+        # because it will resolve paths relative to its true location.
+        # Any symbolic links have to be replicated as well.
+        cd "${browser}"
+        find . -type d -exec mkdir -p "$out"/{} \;
+
+        find . -type f \( -not -name "${browserName}" \) -exec ln -sT "${browser}"/{} "$out"/{} \;
+
+        find . -type f -name "${browserName}" -print0 | while read -d $'\0' f; do
+          cp -P --no-preserve=mode,ownership "${browser}/$f" "$out/$f"
+          chmod a+rwx "$out/$f"
+        done
+
+        # fix links and absolute references
+        cd "${browser}"
+
+        find . -type l -print0 | while read -d $'\0' l; do
+          target=$(readlink "$l" | ${replace}/bin/replace-literal -es -- "${browser}" "$out")
```

Comment:
Should be quoted.

Reply:
Also any particular reason you use "replace-literal` vs. `sed` or `substituteInPlace`?

Reply:
This used to part of an -exec line in find, so substituteInPlace would not have worked.
Sed is about regular expressions, which is just not what we want here.
But the way its written now, we could just use substituteInPlace. I will change that.

Reply:
Ah wait, substituteInPlace can't read from stdin.

Reply:
Why do you need to read from stdin?

Reply:
I want to change the location the symlink points to.
You suggest writing that into an intermediate file, and then running substituteInPlace on that?

Reply:
Oh nevermind, I didn't think this through.

I'd probably still use `sed` for this just because it is more of a "standard tool" and the regex capability wouldn't hurt here, but this should be fine as-is.